14/umbraco-deploy/upgrades/version-specific.md

```diff
@@ -4,83 +4,108 @@ description: >-
 ---
 
 # Version Specific Upgrade Details
-
-This page covers specific upgrade documentation for when migrating to major 13 of Umbraco Deploy.
+This article provides specific upgrade documentation for migrating to Umbraco Deploy version 14.
 
 {% hint style="info" %}
-If you are upgrading to a minor or patch of Deploy you can find the details about the changes in the [release notes](../release-notes.md) article.
+If you are upgrading to a minor or patch version, you can find the details about the changes in the [Release Notes](../release-notes.md) article.
 {% endhint %}
 
 ## Version Specific Upgrade Notes History
-
-Version 13 of Umbraco Deploy has a minimum dependency on Umbraco CMS core of `13.0.0`. It runs on .NET 8.
-
-#### **Breaking changes**
-
-Version 13 contains a number of breaking changes. We don't expect many projects to be affected by them as they are in areas that are not typical extension points. For reference though, the full details are listed here:
-
-#### **Behavior**
-
-* The default value for the configuration option `ResolveUserInTargetEnvironment` was changed to `true`.
-
-#### **Configuration**
-
-* [Configuration of relations](../getting-started/deploy-settings.md#relationtypes) was changed from a list to a dictionary.
+Version 14 of Umbraco Deploy has a minimum dependency on Umbraco CMS core of `14.0.0`. It runs on .NET 8.
+
+### **Breaking changes**
+Version 14 contains a number of breaking changes. We don't expect many projects to be affected by them, as they are in areas that are mostly relevant when extending Deploy to support additional entities and/or property editors. For reference though, the full details are listed here:
+
+#### Async methods
+Asynchronous methods have been added to the following interfaces (in the CMS `Umbraco.Cms.Core.Deploy` namespace):
+- `IContextCache`:
+  - `GetOrCreateAsync(...)`
+- `IDataTypeConfigurationConnector`, `IImageSourceParser`, `ILocalLinkParser` and `IValueConnector`:
+  - `ToArtifactAsync(...)`
+  - `FromArtifactAsync(...)`
+- `IServiceConnector`:
+  - `GetArtifactAsync(...)`
+  - `ProcessInitAsync(...)`
+  - `ProcessAsync(...)`
+  - `ExpandRangeAsync(...)`
+  - `GetRangeAsync(...)`
+
+These methods all have a default implementation that forwards the calls to the synchonous methods (to maintain backwards compatibility). The synchonous methods have been obsoleted though and Deploy will now always call the new asynchronous methods, so implementations should be updated to start using those instead.
+
+Within Deploy, the following base classes and methods have been updated to take advantage of the asynchonous methods:
+- `ValueConnectorBase`, `RecursiveValueConnectorBase`, `DataTypeConfigurationConnectorBase` and `ServiceConnectorBase`: all synchonous methods are obsoleted and cause compiler errors when directly invoked (to avoid potential deadlocks, because they all forward to the asynchonous methods using `GetAwaiter().GetResult()`);
```

Comment:
```suggestion
- `ValueConnectorBase`, `RecursiveValueConnectorBase`, `DataTypeConfigurationConnectorBase` and `ServiceConnectorBase`: all synchronous methods are obsoleted and cause compiler errors when directly invoked (to avoid potential deadlocks, because they all forward to the asynchronous methods using `GetAwaiter().GetResult()`);
```